lib/conrad/emitters/base.rb

```diff
@@ -0,0 +1,45 @@
+require 'conrad/errors'
+
+module Conrad
+  module Emitters
+    # Base class for emitters
+    class Base
```

Comment:
I've been sitting around thinking about this for a while and wondering what your thoughts are on doing this as a Wrapper class rather than as a Base class? It has the benefit of just being added functionality and allowing the same flexibility of emitters + more plug and play in its usage. Its main drawback I can think of is that it would be more verbose and potentially a bit more confusing for people to understand. I dunno.

```ruby
# Example using your threading code
module Conrad
  module Emitters
    class BackgroundHandler
      def initialize(emitter)
        @emitter = emitter
        @queue = Queue.new
        start_background_process
      end

      def call(event)
        queue.push(event)
      end

      private

      attr_reader :queue
      attr_reader :emitter

      def background_process
        Thread.new do
          loop do
            unless queue.empty?
              data = queue.pop
              emitter.call(data)
            end
          end
        end
      end
    end
  end
end

Conrad::Collector.default_emitter = Conrad::Emitters::BackgroundHandler.new(
  Conrad::Emitters::Stdout.new
)
```